### PR TITLE
fix typos

### PR DESCRIPTION
**Pull request by miles-six**
## What ❔

I discovered 24 typos in a total of 22 files. Then I merged them together.

## Why ❔

The code quality is exceptionally high, so sincerely hope my PR can help make this repository more standardized.

## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.
- [ ] Spellcheck has been run via `zk spellcheck`.
- [ ] Linkcheck has been run via `zk linkcheck`.

### CODE REVIEW

**Comment by itegulov** [comment]
Thanks for the contribution. Unfortunately we don't accept PRs that are pure typo corrections but I can see that you used codespell to go through our codebase. We would accept your PR if you could integrate it into our CI by adding it to `zk spellcheck` [here](https://github.com/matter-labs/zksync-era/blob/main/infrastructure/zk/src/spellcheck.ts#L4) (along with `cspell` and `cargo spellcheck` that are already there).

File: core/lib/dal/migrations/20221215094205_prover-job-identity.down.sql [hunk 1]
```diff
@@ -1,2 +1,2 @@
 ALTER TABLE prover_jobs
-    DROP COLUMN proccesed_by;
\ No newline at end of file
+    DROP COLUMN processed_by;
```
Comment (itegulov) [inline core/lib/dal/migrations/20221215094205_prover-job-identity.down.sql:2]: Please exclude the existing migration files that have typos. We can't touch those anymore.

Comment (miles-six) [reply core/lib/dal/migrations/20221215094205_prover-job-identity.down.sql:2]: Yeah,sir,I will do this. Btw,can I add codespell check to workflow to integrate it

Comment (itegulov) [reply core/lib/dal/migrations/20221215094205_prover-job-identity.down.sql:2]: Yes, I left a comment above

Comment (miles-six) [reply core/lib/dal/migrations/20221215094205_prover-job-identity.down.sql:2]: I have added a relatively simple filtering file to exclude certain words. 

If you find it meaningful, you can modify the filtering conditions according to your needs.

skip:Files to exclude from the check
words.txt: Words to exclude from the check

File: .github/workflows/codespell.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
+          restore-keys: ${{ runner.os }}-pip-
+
+      - name: Install prerequisites
+        run: sudo pip install -r ./.codespell/requirements.txt
+
+      - name: Spell check
+        run: codespell --config=./.codespell/.codespellrc
```
Comment (itegulov) [inline .github/workflows/codespell.yml:30]: This shouldn't be its own workflow. Please incorporate this functionality into the `zk` tool as I suggested

File: core/lib/dal/migrations/20221215094205_prover-job-identity.down.sql (hunk 1 above)
Comment (itegulov) [reply core/lib/dal/migrations/20221215094205_prover-job-identity.down.sql:2]: Yes, these should be reverted and added to skip I am guessing

**Comment by popzxc** [comment]
Hello. As @itegulov said, we're currently not accepting typo fix PRs.

**Comment by miles-six** [comment]
> Hello. As @itegulov said, we're currently not accepting typo fix PRs.

Hey sir, as @itegulov said, I will  integrate codespell into  CI. So please don't just closed it, I will do it.

**Comment by popzxc** [comment]
@miles-six We already have both spellcheck and integrated in CI (you can check the GitHub action workflows), so the right course of action would be to introduce changes to the spellcheck dictionary. However, we're currently experiencing an influx of low effort contributions, so right now we are only considering functional contributions.

Typos are obviously bad and deserve fixing, but right now the ratio of external contributions is significantly skewed towards non-functional changes. At the same time, each PR requires support from developers that could've been applied to develop new features.

Hope for your understanding and thanks for the contribution anyway.

**Comment by popzxc** [comment]
@miles-six You can find the relevant CI workflows, check which tools are being used, and track the configuration for these tools.

That said, the likelihood of a PR that *only* introduces changes to the dictionary file getting merged is still extremely low, so, if you want to contribute, it's better to focus on introducing functional changes to the code. Just be sure to check [the contributing guide](https://github.com/matter-labs/zksync-era/blob/main/CONTRIBUTING.md) first to avoid doing work that won't be merged afterwise.

**Pull request closed without merging**